### PR TITLE
Update to velocity 3, Make Proxy Events cancellable

### PR DESCRIPTION
**Pull request by JOO200**
This PR needs https://github.com/NuVotifier/NuVotifier/pull/295 first.
Therefore this is a draft and won't be opened as PR until #295 is merged.

Updated compatibility to Velocity 3 and changed the events on proxy to be cancellable.
This does work under velocity fine.

### CODE REVIEW

File: bungeecord/src/main/java/com/vexsoftware/votifier/bungee/events/VotifierEvent.java [hunk 1]
```diff
@@ -1,23 +1,38 @@
...
+import net.md_5.bungee.api.Callback;
+import net.md_5.bungee.api.event.AsyncEvent;
+import net.md_5.bungee.api.plugin.Cancellable;
 
 /**
  * {@code VotifierEvent} is a custom BungeeCord event class that is sent
  * asynchronously allowing other plugins to listen for votes.
  */
-public class VotifierEvent extends Event {
+public class VotifierEvent extends AsyncEvent<VotifierEvent> implements Cancellable {
```
Comment (astei) [inline bungeecord/src/main/java/com/vexsoftware/votifier/bungee/events/VotifierEvent.java:12]: Thumbs down to making this an `AsyncEvent`, it's a completely broken programming model along with being a breaking change.

I can see some value in having a cancelable event.

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import org.ajoberstar.grgit.Grgit
```
Comment (astei) [inline build.gradle.kts:1]: Why did you convert the build script to Kotlin DSL? That should really be in its own PR.

File: common/src/test/java/com/vexsoftware/votifier/util/ArgsToVoteTest.java [hunk 3]
```diff
@@ -37,72 +27,28 @@ void handlesSingleUsernameTooLong() {
         });
     }
 
-    @ParameterizedTest
```
Comment (astei) [inline common/src/test/java/com/vexsoftware/votifier/util/ArgsToVoteTest.java:40]: Why rewrite these as individual tests? That's not in the scope of this PR.

File: sponge/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+sponge {
+    apiVersion("7.2.0")
+    loader {
+        name(PluginLoaders.JAVA_PLAIN)
+        version("1.0")
+    }
+
+    license("GNU General Public License v3.0")
+
+    plugin("banmanager") {
```
Comment (astei) [inline sponge/build.gradle.kts:28]: Oh, I see. You've haphazardly copied this from somewhere.

**Review by astei — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 2 above)
Comment (JOO200) [reply build.gradle.kts:1]: See #295 and the description of this PR.
I cleaned the build scripts because the existing scripts didn't work. E.g. i was unable to build the universal jar file.

File: common/src/test/java/com/vexsoftware/votifier/util/ArgsToVoteTest.java (hunk 3 above)
Comment (JOO200) [reply common/src/test/java/com/vexsoftware/votifier/util/ArgsToVoteTest.java:40]: #295

File: bungeecord/src/main/java/com/vexsoftware/votifier/bungee/events/VotifierEvent.java (hunk 1 above)
Comment (JOO200) [reply bungeecord/src/main/java/com/vexsoftware/votifier/bungee/events/VotifierEvent.java]: I reverted the changes from the AsyncEvent but kept the Cancellable stuff.

**Comment by JOO200** [comment]
#295 is merged so this is now ready to merge as well.

I tested the changes for Velocity and it works fine on the latest Velocity version.

Currently the version number is 3.0.0-SNAPSHOT. Imo it should be 2.8.0-SNAPSHOT instead as there are no major changes in NuVotifier.
